src/check.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Module for running checks."""
+
+import logging
+from collections.abc import Iterable, Iterator
+from typing import NamedTuple, TypeGuard
+
+from .content import conflicts as content_conflicts
+from .content import diff as content_diff
+from .types_ import AnyAction, UpdateAction
+
+
+class Problem(NamedTuple):
+    """Details about a failed check.
+
+    Attrs:
+        path: Unique identifier for the file and discourse topic with the problem
+        description: A summary of what the problem is and how to resolve it.
+
+    """
+
+    path: str
+    description: str
+
+
+def _is_update_action(action: AnyAction) -> TypeGuard[UpdateAction]:
+    """Check whether an action is an UpdateAction.
+
+    Args:
+        action: The action to check.
+
+    Returns:
+        Whether the action is an UpdateAction.
+    """
+    return isinstance(action, UpdateAction)
+
+
+def _update_action_problem(action: UpdateAction) -> Problem | None:
+    """Get any problem with an update action.
+
+    Args:
+        action: The action to check.
+
+    Returns:
+        None if there is no problem or the problem if there is an issue with the action.
+    """
+    if action.content_change is None:
+        return None
+
+    if (
+        action.content_change.base is None
+        and action.content_change.server == action.content_change.local
+    ):
+        return None
+
+    if action.content_change.base is None:
```

Comment:
This feels the same as actions.py:181

Reply:
It is similar, although in `action.py` the action is actually getting executed. This is like a pre-run check

Reply:
Resolving for now, please let me know if you have further feedback on this